.github/workflows/vast.yaml

```diff
@@ -813,67 +816,98 @@ jobs:
+            ${{ github.event_name == 'push' && format('tenzir/vast-deps:{0}', github.sha) || '' }}
+            ${{ github.event_name == 'release' && 'tenzir/vast-deps:stable' || '' }}
+            ${{ github.event_name == 'release' && format('tenzir/vast-deps:{0}', needs.determine-version.outputs.release-version) || '' }}
+            ${{ github.event_name == 'push' && 'ghcr.io/tenzir/vast-deps:latest' || '' }}
+            ${{ github.event_name == 'push' && format('ghcr.io/tenzir/vast-deps:{0}', github.sha) || '' }}
+            ${{ github.event_name == 'release' && 'ghcr.io/tenzir/vast-deps:stable' || '' }}
+            ${{ github.event_name == 'release' && format('ghcr.io/tenzir/vast-deps:{0}', needs.determine-version.outputs.release-version) || '' }}
+          build-args: |
+            "VAST_BUILD_OPTIONS=${{ env.VAST_BUILD_OPTIONS }}"
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Build and Push Development Docker Image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          target: development
+          push: ${{ github.event_name == 'push' || github.event_name == 'release' }}
+          tags: |
+            ${{ github.event_name == 'push' && 'tenzir/vast-dev:latest' || '' }}
+            ${{ github.event_name == 'push' && format('tenzir/vast-dev:{0}', github.sha) || '' }}
+            ${{ github.event_name == 'release' && 'tenzir/vast-dev:stable' || '' }}
+            ${{ github.event_name == 'release' && format('tenzir/vast-dev:{0}', needs.determine-version.outputs.release-version) || '' }}
+            ${{ github.event_name == 'push' && 'ghcr.io/tenzir/vast-dev:latest' || '' }}
+            ${{ github.event_name == 'push' && format('ghcr.io/tenzir/vast-dev:{0}', github.sha) || '' }}
+            ${{ github.event_name == 'release' && 'ghcr.io/tenzir/vast-dev:stable' || '' }}
+            ${{ github.event_name == 'release' && format('ghcr.io/tenzir/vast-dev:{0}', needs.determine-version.outputs.release-version) || '' }}
+          build-args: |
+            "VAST_BUILD_OPTIONS=${{ env.VAST_BUILD_OPTIONS }}"
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Build and Push Production Docker Image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          target: production
+          push: ${{ github.event_name == 'push' || github.event_name == 'release' }}
+          tags: |
+            ${{ github.event_name == 'push' && 'tenzir/vast:latest' || '' }}
+            ${{ github.event_name == 'push' && format('tenzir/vast:{0}', github.sha) || '' }}
+            ${{ github.event_name == 'release' && 'tenzir/vast:stable' || '' }}
+            ${{ github.event_name == 'release' && format('tenzir/vast:{0}', needs.determine-version.outputs.release-version) || '' }}
+            ${{ github.event_name == 'push' && 'ghcr.io/tenzir/vast:latest' || '' }}
```

Comment:
Note to the reviewer: I'm uploading to ghcr.io as well as (1) that causes us to see the Docker images directly on our main GitHub page, and (2) is easier to download from in subsequent steps that depend on this one.

Reply:
This changed slightly in the meantime because I refactored it for less repetitiveness in the CI.